JMX total number of keys in SelectorManager

strogiyotec opened the pull request:
Added a new JMX property in SelectorManager that reports the total number of keys from all ManagedSelectors
How it was tested? 
Run embedded Jetty
```
        final Server server = new Server(9090);
        final MBeanContainer mbeanContainer = new MBeanContainer(ManagementFactory.getPlatformMBeanServer());
        server.addBeanToAllConnectors(mbeanContainer);
        server.start();
        server.join();

```
Open VisualVM and check a new JMX property exists 
![image](https://github.com/eclipse/jetty.project/assets/20584185/0c2346b0-35c7-4a61-b227-6f1fe96229fa)

strogiyotec commented:
@sbordet  can you take a look

sbordet commented on jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java:
```diff
@@ -139,6 +139,23 @@ protected void execute(Runnable task)
+     *
+     * @return total number of selector keys
+     */
+    @ManagedAttribute(value = "Total number of keys in all Managed Selectors", readonly = true)
```
```suggestion
    @ManagedAttribute(value = "Total number of keys in all selectors", readonly = true)
```

strogiyotec replied:
thanks will do

strogiyotec replied:
done

sbordet commented on jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java:
```diff
@@ -139,6 +139,23 @@ protected void execute(Runnable task)
+     * @return total number of selector keys
+     */
+    @ManagedAttribute(value = "Total number of keys in all Managed Selectors", readonly = true)
+    public int getTotalNumberOfSelectorKeys()
```
```suggestion
    public int getTotalKeys()
```

strogiyotec replied:
done

sbordet reviewed (changes requested):
Please also override `toString()` as:

`String.format("%s@%x[keys=%d]", getClass().getSimpleName(), hashCode(), getTotalKeys())`

strogiyotec committed:
JMX total number of keys in SelectorManager

Added a new JMX property in SelectorManager that reports total number of keys from all ManagedSelectors

strogiyotec commented:
@sbordet  I addressed all the comments please take a look

sbordet committed:
Merge branch 'jetty-10.0.x' into jetty-10.0.x

sbordet commented:
@strogiyotec Thanks!
